src/mono/mono/mini/aot-compiler.c

```diff
@@ -5309,7 +5310,7 @@ MONO_RESTORE_WARNING
 				}
 				mono_reflection_free_custom_attr_data_args_noalloc (decoded_args);
 
-				wrapper = mono_marshal_get_managed_wrapper (method, NULL, 0, error);
+				wrapper = mono_marshal_get_managed_wrapper (method, NULL, 0, acfg->aot_opts.runtime_init_callback != NULL && export_name != NULL, error);
```

Comment:
I'm not a huge fan of passing an extra argument to this marshaling call.  What do you think about making a special entrypoint `mono_marshal_get_managed_wrapper_with_runtime_init` (or a better name).

Reply:
OK, lets add one new function an split the old into one that accepts that argument and one that doesn't, didn't want to add that change all the way since it would mean a new function in the marshaller callback as well, but we can definitely split it so it wont affect all callsites.

Reply:
We don't need to split the marshaller callback.  That's not "user visible".  It's kind of an implementation detail of marshal.c.   But all the places that call `mono_marshal_get_managed_wrappers` are better off with a simpler interface.

Reply:
jupp that is now already done in this PR.